x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/insecure_output_decorator.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package modifiers
+
+import (
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/configuration"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/transpiler"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+)
+
+const (
+	sslKey              = "ssl"
+	verificationModeKey = "verification_mode"
+	sslVerificationKey  = "ssl.verification_mode"
+)
+
+// InjectInsecureOutput injects a verification none into output configuration.
+func InjectInsecureOutput(fleetConfig *configuration.FleetAgentConfig) func(*logger.Logger, *transpiler.AST) error {
+	return func(_ *logger.Logger, rootAst *transpiler.AST) error {
+		// if verification mode is not set abort
+		if fleetConfig == nil ||
+			fleetConfig.Server == nil ||
+			fleetConfig.Server.TLS == nil ||
+			fleetConfig.Server.TLS.VerificationMode == tlscommon.VerifyFull {
```

Comment:
This does seem to have the affect that it will only do something if the Elastic Agent is running Fleet Server, but it will affect all beats.